spark-knn-core/src/main/scala/org/apache/spark/ml/classification/KNNClassifier.scala

```diff
@@ -150,7 +150,8 @@ with KNNModelParams with HasWeightCol with Serializable {
       }
     }
 
-    val merged = transform(dataset, topTree, subTrees)
+    val neighborDataset : RDD[(Long, Array[Row])] = transform(dataset, topTree, subTrees)
```

Comment:
Can you rename this to `neightborRDD` instead since it is not typed as `Dataset`?